Neos.FluidAdaptor/Classes/ViewHelpers/Form/SelectViewHelper.php

```diff
@@ -262,9 +274,14 @@ protected function getOptions()
                             throw new ViewHelper\Exception('Label value for object of class "' . get_class($value) . '" was an object without a __toString() method.', 1247827553);
                         }
                     }
+                } elseif (is_array($value)) {
+                    throw new ViewHelper\Exception(
+                        '$optionLabelField must be provided for array $options',
+                        1602432476053
+                    );
                 } elseif (method_exists($value, '__toString')) {
                     $value = (string)$value;
-                } elseif ($this->persistenceManager->getIdentifierByObject($value) !== null) {
+                } elseif (!$this->persistenceManager->isNewObject($value)) {
```

Comment:
Right:
```
     * Note: this returns an identifier even if the object has not been
     * persisted in case of AOP-managed entities. Use isNewObject() if you need
     * to distinguish those cases.
```
In theory, you could have a reference to a new object in the select box and it will only be persisted after submitting that form (but with that ID)